### PR TITLE
show runs within backfill checkbox

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
Hooks up the "show runs within backfills" checkbox to the backend

## How I Tested These Changes
View with runs in backfills hidden (default view)
Screenshot 2024-09-27 at 2 17 34 PM

View showing runs in backfills 
Screenshot 2024-09-27 at 2 17 25 PM

## Changelog
NOCHANGELOG

### CODE REVIEW

**Comment by jamiedemaria** [comment]
* **#24849** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24849?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jamiedemaria and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: js_modules/dagster-ui/packages/ui-core/src/runs/RunsFilterInput.tsx [hunk 1]
```diff
@@ -104,12 +109,16 @@ export function useQueryPersistedRunFilters(enabledFilters?: RunFilterTokenType[
...
+          const res = tokenizedValuesFromStringArray(q, RUN_PROVIDERS_EMPTY);
+          return res.filter(
             (t) =>
               !t.token || !enabledFilters || enabledFilters.includes(t.token as RunFilterTokenType),
-          ) as RunFilterToken[],
+          ) as RunFilterToken[];
+        },
       }),
       [enabledFilters],
     ),
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/runs/RunsFilterInput.tsx:124]: you should undo these changes. I just did it for debugging

**Review by salazarm — APPROVED** [review]
